### PR TITLE
docs: add component page for tool calls

### PR DESCRIPTION
**Pull request by ccurme**
Note: includes links to API reference pages for ToolCall and other objects that currently don't exist (e.g., https://api.python.langchain.com/en/latest/messages/langchain_core.messages.tool.ToolCall.html#langchain_core.messages.tool.ToolCall).

### CODE REVIEW

**Review by eyurtsev — APPROVED** [review]

File: docs/docs/modules/model_io/chat/function_calling.mdx [hunk 1]
```diff
@@ -1,68 +1,117 @@
...
-guide](/docs/modules/model_io/chat/structured_output/).
-
-Before getting started make sure you have `langchain-core` installed.
+# Tool calling
 
-```python
-%pip install -qU langchain-core langchain-openai
+# Calling Tools
+
+Tool calling is a common feature of LLM applications. A [tool call](https://api.python.langchain.com/en/latest/messages/langchain_core.messages.tool.ToolCall.html#langchain_core.messages.tool.ToolCall)
```
Comment (jacoblee93) [inline docs/docs/modules/model_io/chat/function_calling.mdx:10]: Tautological

File: docs/docs/modules/model_io/chat/function_calling.mdx [hunk 2]
```diff
@@ -1,68 +1,117 @@
...
-
-Before getting started make sure you have `langchain-core` installed.
+# Tool calling
 
-```python
-%pip install -qU langchain-core langchain-openai
+# Calling Tools
+
+Tool calling is a common feature of LLM applications. A [tool call](https://api.python.langchain.com/en/latest/messages/langchain_core.messages.tool.ToolCall.html#langchain_core.messages.tool.ToolCall) 
+represents a call to a specific tool, and includes a name, arguments dict, and
```
Comment (jacoblee93) [inline docs/docs/modules/model_io/chat/function_calling.mdx:11]: Tautological, how about:

```
Tool calling allows a model to respond to a given prompt by generating output that matches a user-defined schema. While the name implies that the model is performing some action, this is actually not the case! The model is merely coming up with the arguments to a tool, and actually running the tool (or not) is up to the user - for example, if you want to [extract output matching some schema](/docs/use_cases/extraction/) from unstructured text, you could give the model an "extraction" tool that takes parameters matching the desired schema, then treat the generated output as your final result.

A tool call includes a name, arguments dict, and an optional identifier.
```

File: docs/docs/modules/model_io/chat/function_calling.mdx [hunk 3]
```diff
@@ -1,68 +1,117 @@
...
-```python
-%pip install -qU langchain-core langchain-openai
+# Calling Tools
+
+Tool calling is a common feature of LLM applications. A [tool call](https://api.python.langchain.com/en/latest/messages/langchain_core.messages.tool.ToolCall.html#langchain_core.messages.tool.ToolCall) 
+represents a call to a specific tool, and includes a name, arguments dict, and 
+(optionally) an identifier. The arguments dict is structured 
+`{argument_name: argument_value}`.
+
+Many LLM providers, including Anthropic, Cohere, Google, Mistral, OpenAI, and others,
```
Comment (jacoblee93) [inline docs/docs/modules/model_io/chat/function_calling.mdx:15]: Make these links

File: docs/docs/modules/model_io/chat/function_calling.mdx [hunk 4]
```diff
@@ -1,68 +1,117 @@
 ---
 sidebar_position: 2
-title: Function calling
+title: Tool calling
```
[the remark this answers is not in the document]
Comment (baskaryan) [reply docs/docs/modules/model_io/chat/function_calling.mdx:3]: and maybe we add a note callout at the very top saying that we mean them interchangeably, and that we default to treating all models as though they can return multiple tool calls in each message

File: docs/docs/modules/model_io/chat/function_calling.mdx [hunk 5]
```diff
@@ -72,78 +121,178 @@ import ChatModelTabs from "@theme/ChatModelTabs";
...
-reference](https://api.python.langchain.com/en/latest/chat_models/langchain_openai.chat_models.base.ChatOpenAI.html#langchain_openai.chat_models.base.ChatOpenAI.bind_tools).
+```python
+print(type(gathered.tool_calls[0]["args"]))
+```
+
+```text
+<class 'dict'>
+```
 
 ## Defining functions schemas
```
Comment (baskaryan) [inline docs/docs/modules/model_io/chat/function_calling.mdx:297]: shoudl we delete the rest of this doc, user shouldn't have to think about function schemas anymore right?

File: docs/docs/modules/model_io/chat/function_calling.mdx (hunk 4 above)
[the remark this answers is not in the document]
Comment (jacoblee93) [reply docs/docs/modules/model_io/chat/function_calling.mdx:3]: Would be a bit long in the sidebar but down to have a longer note at the top

File: docs/docs/modules/model_io/chat/function_calling.mdx (hunk 5 above)
Comment (jacoblee93) [reply docs/docs/modules/model_io/chat/function_calling.mdx:297]: @ccurme  added it, I think dto underscore why the standard interface is nice. Don't feel super strongly though

File: docs/docs/modules/model_io/chat/function_calling.mdx (hunk 3 above)
Comment (ccurme) [reply docs/docs/modules/model_io/chat/function_calling.mdx]: thanks.

for google link I almost linked to google.com

**Pull request merged**